contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,358 @@
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        if (_getSharesBalance() != 0) return true;
+
+        IRyskLiquidityPool.DepositReceipt
+            memory depositReceipt = _getDepositReceipt();
+
+        if (depositReceipt.amount != 0 || depositReceipt.unredeemedShares != 0)
+            return true;
+
+        IRyskLiquidityPool.WithdrawalReceipt
+            memory withdrawalReceipt = _getWithdrawalReceipt();
+
+        return withdrawalReceipt.shares != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice this also includes shares from the pending withdrawal and shares yet to be minted for pending deposits
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        IRyskLiquidityPool.DepositReceipt
+            memory depositReceipt = _getDepositReceipt();
+        IRyskLiquidityPool.WithdrawalReceipt
+            memory withdrawalReceipt = _getWithdrawalReceipt();
+        uint256 currentPricePerShare = ryskLqPool.withdrawalEpochPricePerShare(
+            ryskLqPool.withdrawalEpoch()
+        );
+
+        // shares for pending deposit (not yet minted or not yet updated on the deposit receipt as unredeemed)
+        uint256 amountInPendingDepositShares = _getAmountInSharesForPendingDeposit(
+                depositReceipt.epoch,
+                depositReceipt.amount,
+                currentPricePerShare
+            );
+
+        // shares marked for withdrawal that are now owned by the pool itself
+        uint256 amountInPendingWithdrawalShares = _sharesToUnderlying(
+            withdrawalReceipt.shares,
+            ryskLqPool.withdrawalEpochPricePerShare(withdrawalReceipt.epoch)
+        );
+
+        // unredeemed shares are not counted in the strategy's balance
+        // because they are not yet claimed (they are still owned by the pool)
+        uint256 amountInUnredeemedShares = _sharesToUnderlying(
+            depositReceipt.unredeemedShares,
+            currentPricePerShare
+        );
+
+        // redeemed shares that are owned by the strategy
+        uint256 amountInRedeemedShares = _sharesToUnderlying(
+            _getSharesBalance(),
+            currentPricePerShare
+        );
+
+        return
+            amountInPendingDepositShares +
+            amountInPendingWithdrawalShares +
+            amountInUnredeemedShares +
+            amountInRedeemedShares;
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+        ryskLqPool.deposit(balance);
```

Comment:
If the `deposit` returns true, don't you need to check the return value here?

Reply:
no, because a call to `deposit` doesn't ever return false, it reverts when something goes wrong